flink-contrib/flink-storm-examples/src/main/java/org/apache/flink/storm/util/FiniteFileSpout.java

```diff
@@ -32,46 +23,17 @@
 public class FiniteFileSpout extends FileSpout implements FiniteSpout {
 	private static final long serialVersionUID = -1472978008607215864L;
 
-	private String line;
-	private boolean newLineRead;
-
 	public FiniteFileSpout() {}
 
 	public FiniteFileSpout(String path) {
 		super(path);
 	}
 
-	@SuppressWarnings("rawtypes")
-	@Override
-	public void open(final Map conf, final TopologyContext context, final SpoutOutputCollector collector) {
-		super.open(conf, context, collector);
-		newLineRead = false;
-	}
-
-	@Override
-	public void nextTuple() {
-		this.collector.emit(new Values(line));
-		newLineRead = false;
-	}
-
 	/**
 	 * Can be called before nextTuple() any times including 0.
 	 */
 	@Override
 	public boolean reachedEnd() {
-		try {
-			readLine();
-		} catch (IOException e) {
-			throw new RuntimeException("Exception occured while reading file " + path);
-		}
-		return line == null;
+		return finished;
 	}
```

Comment:
Following your argument, we should also remove the `FiniteSpout` functionality, because it is not included in Storm as well. I'll revert this but I don't understand your argument.